components/security/core/authentication.rst

```diff
@@ -0,0 +1,229 @@
+
+.. code-block:: php
+
+    use Symfony\Component\Security\Core\Encoder\EncoderFactory;
+    use Symfony\Component\Security\Core\Encoder\PlaintextPasswordEncoder;
+
+    $encoderFactory = new EncoderFactory(array(
+        // this is the FQCN to the user object used by the in memory user provider
+        'Symfony\Component\Security\Core\User\User' => new PlaintextPasswordEncoder(),
+    ));
+
+Now the user is retrieved and matched against the submitted password, the dao
+provider has to do some final checks. For instance, if the user is not banned
+or if the account is not expired. Such checks are done by
+:class:`user checkers <Symfony\\Component\\Security\\Core\\User\\UserCheckerInterface>`::
+
+    use Symfony\Component\Security\Core\User\UserChecker;
+
+    $userChecker = new UserChecker();
+
+Everything is now set-up and ready to be combined into the dao authentication
+provider!
+
+.. code-block::
+
+    use Symfony\Component\Security\Core\Authentication\Provider\DaoAuthenticationProvider;
+    use Symfony\Component\Security\Core\Encoder\EncoderFactory;
+    use Symfony\Component\Security\Core\Encoder\PlaintextPasswordEncoder;
+    use Symfony\Component\Security\Core\User\InMemoryUserProvider;
+    use Symfony\Component\Security\Core\User\UserChecker;
+
+    $userProvider = new InMemoryUserProvider(array(
+        'wouter' => array('password' => 'pa$$'),
+    ));
+
+    $encoderFactory = new EncoderFactory(array(
+        // this is the FQCN to the user object used by the in memory user provider
+        'Symfony\Component\Security\Core\User\User' => new PlaintextPasswordEncoder(),
+    ));
+
+    $userChecker = new UserChecker();
+
+    $daoAuthProvider = new DaoAuthenticationProvider(
+        $userProvider,
+        $userChecker,
+        'default',      // used to group multiple providers
+        $encoderFactory
+    );
+
+You can now inject the dao authentication provider in your authentication
+manager::
+
+    // ...
+    $anonymousAuthProvider = new AnonymousAuthenticationProvider('a_custom_key');
+
+    $authManager = new AuthenticationProviderManager(array(
+        $daoAuthProvider,
+        $anonymousAuthProvider,
+    ));
+
+Now, you can test this out. As both the dao and anonymous authentication
+providers are configured, you can now authenticate an anonymous user as well as
+a real member::
+
+    // ...
+    use Symfony\Component\Security\Core\Authentication\Token\UsernamePasswordToken;
+
+    $token = new UsernamePasswordToken(
+        'wouter', // username
+        'pa$$',   // password
+        'default' // provider key (used to determine which providers are targetted)
+    );
+
+    try {
+        $authenticatedToken = $authManager->authenticate($token);
+    } catch (AuthenticationException $e) {
+        // ... bummer, user submitted a wrong username/password or has not been registered
+    }
+
+    // ... user is authenticated and a real member!
+
+Other Authentication Providers
```

Comment:
My guess is that readers expect thes three auth providers to be explained in detail as the previous ones.